packages/language-php/grammars/tree-sitter/queries/highlights.scm

```diff
@@ -233,19 +233,56 @@
 
 (class_declaration (name) @entity.name.type.class.php)
 
+(class_declaration (base_clause (name) @entity.other.inherited-class.php))
+(class_declaration (class_interface_clause (name) @entity.other.implemented-interface.php))
+
+; usage or consumption of traits
+(use_declaration (name) @entity.name.type.trait.php)
+; in use lists, "bar" in `Foo::bar` is a method, not a constant
+(use_list
+  (_
+    (class_constant_access_expression (name) @support.other.function.method.php .)
+    (name) @support.other.function.method.php
+  )
+  (#set! capture.final true)
+)
+
+; the "Foo" and "bar" in `Foo::bar`
+(class_constant_access_expression . (name) @support.class.php)
+(class_constant_access_expression (name) @variable.other.property.php .)
+
+; the "Foo" and "$bar" in Foo::$bar
+(scoped_property_access_expression
+  scope: (name) @support.class.php
+  name: (variable_name) @variable.other.property.static.php
+  (#set! capture.final true))
+
 ; Static method calls.
 (scoped_call_expression
+  scope: (name) @support.class.php
   name: (name) @support.other.function.method.static.php)
 
 (member_call_expression
   name: (name) @support.other.function.method.php)
 
-
 ; TRAITS
 ; ======
 
 (trait_declaration (name) @entity.name.type.trait.php)
 
+; INTERFACES
+; ======
+
+(interface_declaration (name) @entity.name.type.interface.php)
+
+; ENUMS
+; =====
+
+(enum_declaration
```

Comment:
Docs for enums: https://www.php.net/manual/en/language.enumerations.backed.php

Reply:
We'll also need this for references to enums — e.g., `Suit::Spades`:

```scm
(class_constant_access_expression
  (name) @support.class.php
  (#is? test.first))

(class_constant_access_expression
  (name) @constant.other.enum.php
  (#is? test.last))
```

We want `Suit` to be `support` (no way to distinguish it from a class method being called, so let's just match the `support.class.php` we see elsewhere) and `Spades` to be `constant`. Ideally, there'd be field names so we don't have to do first/last tests, but this'll do.

It falls down if `Foo::Bar::Baz` is an enum lookup, but it's a start.

Reply:
Scratch that — I can use anchors instead of the predicates in the above example. I'm not used to using anchors because they don't work on anonymous nodes.